### PR TITLE
 Make nce support more distribution.

### PR DESCRIPTION
**Pull request by wanghaoshuang**

### CODE REVIEW

**Review by seiriosPlus — DISMISSED** [review]
LGTM!
But The API needs @panyx0718 @typhoonzero @qingqing01 to review.

File: paddle/fluid/API.spec [hunk 1]
```diff
@@ -97,7 +97,7 @@ paddle.fluid.layers.warpctc ArgSpec(args=['input', 'label', 'blank', 'norm_by_ti
 paddle.fluid.layers.sequence_reshape ArgSpec(args=['input', 'new_dim'], varargs=None, keywords=None, defaults=None)
 paddle.fluid.layers.transpose ArgSpec(args=['x', 'perm', 'name'], varargs=None, keywords=None, defaults=(None,))
 paddle.fluid.layers.im2sequence ArgSpec(args=['input', 'filter_size', 'stride', 'padding', 'input_image_size', 'out_stride', 'name'], varargs=None, keywords=None, defaults=(1, 1, 0, None, 1, None))
-paddle.fluid.layers.nce ArgSpec(args=['input', 'label', 'num_total_classes', 'sample_weight', 'param_attr', 'bias_attr', 'num_neg_samples', 'name'], varargs=None, keywords=None, defaults=(None, None, None, None, None))
+paddle.fluid.layers.nce ArgSpec(args=['input', 'label', 'num_total_classes', 'sampler', 'custom_dist', 'sample_weight', 'param_attr', 'bias_attr', 'num_neg_samples', 'seed', 'name'], varargs=None, keywords=None, defaults=('uniform', None, None, None, None, None, 0, None))
```
Comment (typhoonzero) [inline paddle/fluid/API.spec]: Maybe add new arguments at the end of the arg list to keep API compatibility.

Comment (wanghaoshuang) [reply paddle/fluid/API.spec]: Thx. fixed.

**Review by typhoonzero — APPROVED** [review]
LG API!

**Review by qingqing01 — APPROVED** [review]

**Pull request merged**